### ISSUE TITLE
sub class method params assignability check doesn't give error with string vs literals

### ISSUE THREAD

**Issue by LabhanshAgrawal** [microsoft/TypeScript#39571]
**TypeScript Version:**  3.9.2 (also tested in nightly)

**Search Terms:**
TS2416 literal
subclass methods literals

**Code**

```ts
class foo {
  fn(x: string) {
    return x;
  }
}

class bar extends foo {
  fn(x: 'a' | 'b' | 'c') {
    return '';
  }
}

class baz extends foo {
  fn(x: 'a' | 'b' | 'c' | undefined) {
    return '';
  }
}
```

**Expected behavior:**

Getting an error `Property 'fn' in type <sub class> is not assignable to the same property in base type 'foo'.`
As
```
Types of parameters 'x' and 'x' are incompatible.
      Type 'string' is not assignable to type <corresponding param type of fn in sub class>.
```
for both classes bar and baz

**Actual behavior:**

The error is only shown on class `baz`. It should also be there on `bar` as `string` is not assignable to `'a' | 'b' | 'c'` also.

**Playground Link:** [playground](https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=17&pc=2#code/MYGwhgzhAEBmD29oG8BQ04DsAUAPAXNBAC4BOAlpgOYCUK6G0pApsQK6mbS4DcDAvqkGpQkGACMwpaM1zFmmACYwESNBlg4C0AORgd0AD67xB4zuA666xi3addOvhkHDRUaJIBeMuQuVwiPQaWoR6ZiYRFhFsSsywlMyK1gwYdhxcOk4CQkA)

**Related Issues:** couldn't find (#35151 came up in search and looked similar but it was a different issue.)

**Comment by RyanCavanaugh** [comment]
Class method types are compared bivariantly; see comments in https://github.com/microsoft/TypeScript/pull/18654
